usr/libexec/helper-scripts/accountctl.sh

```diff
@@ -0,0 +1,352 @@
+## Output: None
+## Return: chpasswd
+## Usage: disable_pass USER
+## Example: disable_pass user
+disable_pass(){
+  log info "${FUNCNAME[0]} $*"
+  has chpasswd
+  local user
+  user="${1:-}"
+  is_user "${user}"
+  if is_pass_disabled "${user}"; then
+    return 0
+  fi
+  pass="$(get_pass "${user}")"
+  chpasswd -c NONE <<< "${user}:*${pass}"
+}
+
+
+## Description: Enable a password.
+## Output: None
+## Return: chpasswd
+## Usage: enable_pass USER
+## Example: enable_pass user
+enable_pass(){
+  log info "${FUNCNAME[0]} $*"
+  has chpasswd
+  local user
+  user="${1:-}"
+  is_user "${user}"
+  if ! is_pass_disabled "${user}"; then
+    return 0
+  fi
+  pass="$(get_clean_pass "${user}" '*')"
+  chpasswd -c NONE <<< "${user}:${pass}"
+}
+
+
+## Description: Transform field name to index per database
+## Output: Field index
+## Return: Exit 1 if entry does not exist
+## Usage: get_field DB FIELD
+## Example: get_field passwd pass
+get_field(){
+  log info "${FUNCNAME[0]} $*"
+  local db field index
+  db="${1:-}"
+  field="${2:-}"
+  index=""
+  case "${db}" in
+    pass|shadow) is_user "${user}" || return 1;;
+    group|gshadow) is_group "${user}" || return 1;;
+  esac
+  case "${db}" in
+    passwd)
+      case "${field}" in
+        pass) index=2;;
+        uid) index=3;;
+        gid) index=4;;
+        comment) index=5;;
+        home) index=6;;
+        shell) index=7;;
+      esac
+      ;;
+    shadow)
+      case "${field}" in
+        pass) index=2;;
+        last-pass-change) index=3;;
+        min-pass-age) index=4;;
+        max-pass-age) index=5;;
+        warn-pass-age) index=6;;
+        lock-pass-age) index=7;;
+        expiration-date) index=7;;
+      esac
+      ;;
+    group)
+      case "${field}" in
+        pass) index=2;;
+        gid) index=3;;
+        members) index=4;;
+      esac
+      ;;
+    gshadow)
+      case "${field}" in
+        pass) index=2;;
+        admins) index=3;;
+        members) index=4;;
+      esac
+      ;;
+    "") log error "No database provided"; return 1;;
+    *) log error "Unsupported database: '${db}'"; return 1;;
+  esac
+  if test -z "${field}"; then
+    log error "Empty ${db} field provided"
+    return 1
+  fi
+  if test -z "${index}"; then
+    log error "Unsupported ${db} field: '${field}'"
+    return 1
+  fi
+  printf '%s' "${index}"
+}
+
+
+## Description: Get database entry
+## Output: Entry
+## Return: getent if database doesn't exist
+## Usage: get_entry USER DATABASE FIELD
+## Example: get_entry user passwd shell
+get_entry(){
+  log info "${FUNCNAME[0]} $*"
+  has getent
+  local user db field index
+  local -a entry
+  user="${1:-}"
+  db="${2:-}"
+  field="${3:-}"
+  index="$(get_field "${db}" "${field}")"
+  IFS=":" read -ra entry <<< "$(getent -- "${db}" "${user}")"
+  ## Array index will be smaller in case the last field is empty.
+  if test ${#entry[@]} -eq $((index-1)); then
+    printf '%s' ""
+  else
+    printf '%s' "${entry[$((index-1))]}"
+  fi
```

Comment:
I think it would improve readability if `get_field` just returned zero-indexed values rather than having to translate the values to zero-indexed by subtracting one here.